app/server/modules/lifecycle/config-export.controller.ts

```diff
@@ -0,0 +1,380 @@
+}
+
+/** Get filter options from request query params */
+function getFilterOptions(c: Context): FilterOptions {
+	return { id: c.req.query("id"), name: c.req.query("name") };
+}
+
+/**
+ * Process secrets in an object based on the secrets mode.
+ * Automatically detects encrypted fields using cryptoUtils.isEncrypted.
+ */
+async function processSecrets(
+	obj: Record<string, unknown>,
+	secretsMode: SecretsMode
+): Promise<Record<string, unknown>> {
+	if (secretsMode === "encrypted") {
+		return obj;
+	}
+
+	const result = { ...obj };
+
+	for (const [key, value] of Object.entries(result)) {
+		if (typeof value === "string" && cryptoUtils.isEncrypted(value)) {
+			if (secretsMode === "exclude") {
+				delete result[key];
+			} else if (secretsMode === "cleartext") {
+				try {
+					result[key] = await cryptoUtils.decrypt(value);
+				} catch (err) {
+					logger.warn(`Failed to decrypt field "${key}": ${err instanceof Error ? err.message : String(err)}`);
+					delete result[key];
+				}
+			}
+		} else if (Array.isArray(value)) {
+			result[key] = await Promise.all(
+				value.map(async (item) =>
+					item && typeof item === "object" && !Array.isArray(item)
+						? processSecrets(item as Record<string, unknown>, secretsMode)
+						: item
+				)
+			);
+		} else if (value && typeof value === "object") {
+			result[key] = await processSecrets(value as Record<string, unknown>, secretsMode);
+		}
+	}
+
+	return result;
+}
+
+/** Clean and process an entity for export */
+async function exportEntity(
+	entity: Record<string, unknown>,
+	params: ExportParams
+): Promise<Record<string, unknown>> {
+	const cleaned = omitKeys(entity, params.excludeKeys);
+	return processSecrets(cleaned, params.secretsMode);
+}
+
+/** Export multiple entities */
+async function exportEntities<T extends Record<string, unknown>>(
+	entities: T[],
+	params: ExportParams
+): Promise<Record<string, unknown>[]> {
+	return Promise.all(entities.map((e) => exportEntity(e as Record<string, unknown>, params)));
+}
+
+// ============================================================================
+// Data Fetchers with Filtering
+// ============================================================================
+
+async function fetchVolumes(filter: FilterOptions): Promise<FetchResult<typeof volumesTable.$inferSelect>> {
+	if (filter.id) {
+		const id = Number.parseInt(filter.id, 10);
+		if (Number.isNaN(id)) return { error: "Invalid volume ID", status: 400 };
+		const result = await db.select().from(volumesTable).where(eq(volumesTable.id, id));
+		if (result.length === 0) return { error: `Volume with ID '${filter.id}' not found`, status: 404 };
+		return { data: result };
+	}
+	if (filter.name) {
+		const result = await db.select().from(volumesTable).where(eq(volumesTable.name, filter.name));
+		if (result.length === 0) return { error: `Volume '${filter.name}' not found`, status: 404 };
+		return { data: result };
+	}
+	return { data: await db.select().from(volumesTable) };
+}
```

Comment:
We have a volumes.service to do this. This code should not be in the config-export.controller